Rebuild issues with core 3.4.0 and compat 3.3.0

Seems to build the first time OK, but then after changing code (causing a rebuild) this error occurs in the browser. Getting it to work again requires deleting node_modules / reinstall. Using pnpm.

```
Uncaught Error: Assertion Failed: [ember-data-factory-guy] FactoryGuy#setStore needs a valid store instance. You passed in [<teacher-ui-client@service:store::ember113>]
```

![Screenshot 2023-11-13 at 9 07 41 pm](https://github.com/embroider-build/embroider/assets/1049837/bbd147ed-9587-4d92-a295-f572cd89be8b)

tests seem to show a bit more detail
![Screenshot 2023-11-13 at 9 10 53 pm](https://github.com/embroider-build/embroider/assets/1049837/0216eaa5-a956-459d-8ede-ed278945cb84)

I have also encountered a similar issue with mirage in some other addons I use

```
[2023-11-13T03:54:18Z] ERROR in ../rewritten-packages/ember-cli-mirage.41f0748a/node_modules/ember-cli-mirage/_app_/mirage/config.js 2:0-81

[2023-11-13T03:54:18Z] Module not found: Error: Can't resolve '@blakeelearning/app-refresher/mirage/config' in '/workdir/packages'

[2023-11-13T03:54:18Z] @ ./assets/test-app.js 342:13-49

[2023-11-13T03:54:18Z]

[2023-11-13T03:54:18Z] ERROR in ../rewritten-packages/ember-cli-mirage.41f0748a/node_modules/ember-cli-mirage/_app_/mirage/config.js 3:0-93

[2023-11-13T03:54:18Z] Module not found: Error: Can't resolve '@blakeelearning/student-events/mirage/config' in '/workdir/packages'

[2023-11-13T03:54:18Z] @ ./assets/test-app.js 342:13-49

[2023-11-13T03:54:18Z]

[2023-11-13T03:54:18Z] ERROR in ../rewritten-packages/ember-cli-mirage.41f0748a/node_modules/ember-cli-mirage/_app_/mirage/config.js 4:0-86

[2023-11-13T03:54:18Z] Module not found: Error: Can't resolve '@blakeelearning/student-progress/mirage/config' in '/workdir/packages'

[2023-11-13T03:54:18Z] @ ./assets/test-app.js 342:13-49

[2023-11-13T03:54:18Z]

[2023-11-13T03:54:18Z] ERROR in ../rewritten-packages/ember-cli-mirage.41f0748a/node_modules/ember-cli-mirage/_app_/mirage/factories/student-progress.js 1:0-93

[2023-11-13T03:54:18Z] Module not found: Error: Can't resolve '@blakeelearning/student-progress/mirage/factories/student-progress' in '/workdir/packages'

[2023-11-13T03:54:18Z] @ ./assets/test-app.js 348:13-69

[2023-11-13T03:54:18Z]

[2023-11-13T03:54:18Z] ERROR in ../rewritten-packages/ember-cli-mirage.41f0748a/node_modules/ember-cli-mirage/_app_/mirage/serializers/student-progress.js 1:0-95

[2023-11-13T03:54:18Z] Module not found: Error: Can't resolve '@blakeelearning/student-progress/mirage/serializers/student-progress' in '/workdir/packages'

[2023-11-13T03:54:18Z] @ ./assets/test-app.js 354:13-71

[2023-11-13T03:54:18Z]

[2023-11-13T03:54:18Z] 5 errors have detailed information that is not shown.

[2023-11-13T03:54:18Z] Use 'stats.errorDetails: true' resp. '--stats-error-details' to show it.

[2023-11-13T03:54:18Z]

[2023-11-13T03:54:18Z] webpack 5.89.0 compiled with 5 errors and 4 warnings in 7277 ms

[2023-11-13T03:54:18Z] cleaning up

[2023-11-13T03:54:18Z] cleaning up...

[2023-11-13T03:54:18Z] Build Error (PackagerRunner) in ../rewritten-packages/ember-cli-mirage.41f0748a/node_modules/ember-cli-mirage/_app_/mirage/config.js

[2023-11-13T03:54:18Z]

[2023-11-13T03:54:18Z] Module not found: Error: Can't resolve '@blakeelearning/app-refresher/mirage/config' in '/workdir/packages'
```

---

Going to guess it's likely to do with https://github.com/embroider-build/embroider/pull/1627

--- a/packages/core/src/ember-html.ts
+++ b/packages/core/src/ember-html.ts
@@ -62,11 +62,8 @@ class Placeholder {
 
   clear() {
     let { start, end, parent } = this;
-    let target = start.nextSibling;
-
-    while (target && target !== end) {
-      parent.removeChild(target);
-      target = target.nextSibling;
+    while (start.nextSibling && start.nextSibling !== end) {
+      parent.removeChild(start.nextSibling);
     }
   }
 

